src/components/CompareResults/beta/DownloadButton.tsx

```diff
@@ -16,78 +19,42 @@ const styles = {
   }),
 };
 
-interface GroupedResults {
-  [key: string]: ResultObject[];
-}
+function DownloadButton() {
+  const [blobUrl, setBlobUrl] = useState<string | null>(null);
 
-interface ResultObject {
-  header_name: string;
-  new_rev: string;
-}
+  const results = useAppSelector(selectStringifiedJsonResults);
 
-function DownloadButton() {
-  let fileName = 'perf-compare-all-revisions.json';
-  const results = useAppSelector((state) => {
+  const fileName = useAppSelector((state) => {
     if (
       state.comparison.activeComparison ===
       Strings.components.comparisonRevisionDropdown.allRevisions
     ) {
-      return state.compareResults.data;
+      return 'perf-compare-all-revisions.json';
     } else {
-      fileName = `perf-compare-${truncateHash(
+      return `perf-compare-${truncateHash(
         state.comparison.activeComparison,
       )}.json`;
-      return {
-        [state.comparison.activeComparison]:
-          state.compareResults.data[state.comparison.activeComparison],
-      };
     }
   });
-
-  const formatDownloadData = (
-    data: Record<string, ResultObject[]>,
-  ): object[] => {
-    const groupNames = Object.keys(data);
-    const transformedGroups: object[] = [];
-
-    groupNames.forEach((groupName) => {
-      const groupedResults = data[groupName].reduce(
-        (grouped: GroupedResults, result: ResultObject) => {
-          if (!grouped[result.header_name]) {
-            grouped[result.header_name] = [];
-          }
-          grouped[result.header_name].push(result);
-          return grouped;
-        },
-        {},
-      );
-
-      const transformedGroupEntries = Object.keys(groupedResults).map(
-        (header_name) => ({
-          [`${header_name} ${truncateHash(
-            groupedResults[header_name][0].new_rev,
-          )}`]: groupedResults[header_name],
-        }),
-      );
-
-      transformedGroups.push(...transformedGroupEntries);
-    });
-    return transformedGroups;
-  };
-
+  useLayoutEffect(() => {
+    if (results) {
+      const blob = new Blob([results], { type: 'application/json' });
+      const newBlobUrl = URL.createObjectURL(blob);
+      setBlobUrl(newBlobUrl);
+    }
+    return () => {
+      if (blobUrl !== null) {
+        URL.revokeObjectURL(blobUrl);
+      }
+    };
+  }, [results]);
   return (
     <div className={styles.downloadButton}>
-      <Button
-        disabled={!results}
-        href={`data:text/json;charset=utf-8,${encodeURIComponent(
-          JSON.stringify(formatDownloadData(results)),
-        )}`}
-        download={fileName}
-      >
+      <Button href={blobUrl ?? ''} download={fileName}>
```

Comment:
BTW it would be good to test that we actually get a blob url eventually, maybe you'll need to wait until it's rendered (maybe something like `await screen.findByText("Download JSON", { selector: "[href^=blob:]")`).

Reply:
Still waiting for this :-) thanks